core/src/main/java/org/opensearch/sql/executor/pagination/Cursor.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.sql.executor.pagination;
+
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+
+@EqualsAndHashCode
+public class Cursor {
```

Comment:
This new class is actually a `String`? I don't see any custom behavior other than that.

Reply:
For the moment, yes but I believe it's an implementation detail worth abstracting.

Reply:
👍 I'll take a look.